src/model.rs

```diff
@@ -313,47 +327,52 @@ mod tests {
     use super::*;
     use crate::status::Status;
 
-     #[test]
+    #[test]
     fn call_solve_without_problem() {
         assert!(Model::new().unwrap().solve().is_err());
     }
 
     #[test]
-    fn solution_without_problem() {
-        let mut model = Model::new().unwrap();
-        let sol = model.get_best_sol();
-        sol.get_obj_val();
-    }
-
-    #[test]
-    fn drop_problem_before_solution() {
-        let sol = {
-            let mut model = Model::new().unwrap();
-            model.hide_output().unwrap();
-            model.include_default_plugins().unwrap();
-            model.read_prob("data/test/simple.lp").unwrap();
-            model.solve().unwrap();
-            model.get_best_sol()
-        };
-        assert_eq!(sol.get_obj_val(), 200.);
+    fn call_solve_on_empty_problem() {
+        assert!(Model::new().unwrap().solve().is_err());
     }
 
     #[test]
-    fn drop_variable_after_problem() {
-        let mut model = Model::new().unwrap();
-        let var_id = model.add_var(0., 0., 0., "", VarType::Binary).unwrap();
-        let var = model.get_var(var_id).unwrap();
-        drop(model);
-        drop(var);
-    }
+    fn solution_without_problem() {
+        let model = Model::new().unwrap();
+        let sol = model.get_best_sol();
+        assert!(sol.is_none());
+    }
+
+    // #[test] does not compile anymore
+    // fn drop_problem_before_solution() {
+    //     let sol = {
+    //         let mut model = Model::new().unwrap();
+    //         model.hide_output().unwrap();
+    //         model.include_default_plugins().unwrap();
+    //         model.read_prob("data/test/simple.lp").unwrap();
+    //         model.solve().unwrap();
+    //         model.get_best_sol()
+    //     };
+    //     assert_eq!(sol.get_obj_val(), 200.);
+    // }
+
+    // #[test]  does not compile anymore
+    // fn drop_variable_after_problem() {
+    //     let mut model = Model::new().unwrap();
+    //     let var_id = model.add_var(0., 0., 0., "", VarType::Binary).unwrap();
+    //     let var = model.get_var(var_id).unwrap();
+    //     drop(model);
+    //     drop(var);
+    // }
```

Comment:
No, you don't need to assert that tests using an outdated API do not compile. You want to write new tests against your new updated safer API to test all edge cases in it and guarantee it is safe in all possible usages.

Reply:
It's not an outdated API, it will not compile because `model` cannot be moved outside this function scope when a reference to it through `var` is still used.

Reply:
Yes, and that's an api change. What we need to do is write new tests that compile with this change, but still exert unusual code paths.